webdriver/tests/support/helpers.py

```diff
@@ -60,24 +60,37 @@ def _restore_window_state(session):
             session.window.size = defaults.WINDOW_SIZE
 
     @ignore_exceptions
-    def _restore_windows(session):
+    async def _restore_windows(session):
         """Close superfluous windows opened by the test.
 
         It will not end the session implicitly by closing the last window.
+        If available, uses WebDriver BiDi session. Otherwise, uses the WebDriver Classic.
         """
-        current_window = session.window_handle
 
-        for window in _windows(session, exclude=[current_window]):
-            session.window_handle = window
-            if len(session.handles) > 1:
-                session.window.close()
+        if session.bidi_session:
+            response = await (
+                await session.bidi_session.send_command("browsingContext.getTree", {
+                    "maxDepth": 1
+                }))
+            for context in response["contexts"][1:]:
+                await (
```

Comment:
double await?

Reply:
yes, as `send_command` returns a future of future. AFAIK, unlike JS, in Python the nested promises are not waited automatically.

Reply:
wdyt about splitting into two lines?

```
future_response = await session.bidi_session.send_command("browsingContext.close", {
    "context": context["context"]
}))
await future_response
```

Reply:
I don't mind making any stylistic changes if we agree on the approach

Reply:
Agreed Alex's version is slightly more readable, but if you go with it then please rename `future_response` to `browsing_context_close_future` or similar.

Reply:
done